### PR TITLE
docs: enable opensearch

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Adds an [OpenSearch](https://developer.mozilla.org/en-US/docs/Web/OpenSearch) description file to the docs:
image

It only registers *sometimes* in chromium-based browsers and no-one really seems to know why, I think at this point I've spent more time looking at the chromium bugtracker and other projects' GitHub issues than actually implementing this.
It works fine in Firefox (see screenshot) :shrug: 

sphinx config docs: https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_use_opensearch

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -109,6 +110,9 @@
 release = version
 
 
+_is_readthedocs = bool(os.getenv("READTHEDOCS"))
```
Comment (onerandomusername) [inline docs/conf.py:113]:
```suggestion
_IS_READTHEDOCS = bool(os.getenv("READTHEDOCS"))
```
and everywhere that uses the variable

Comment (shiftinv) [reply docs/conf.py]: https://github.com/DisnakeDev/disnake/pull/859/commits/3b1624be1a14325f63753b362e9bcf82f36a8820

**Review by toolifelesstocode — APPROVED** [review]
Seems to work. However it does not work on Opera/Opera GX, I'm _think_ it's a browser issue, not a PR issue 🙂

Test:

**Comment by xlvlocl** [comment]
I am in <#828758532467195935>  hisotry

**Review by onerandomusername — APPROVED** [review]

**Comment by onerandomusername** [comment]
previously reviewed, forgot to comment, lgtm

**Pull request merged**